src/main.cpp

```diff
@@ -1756,7 +1756,12 @@ bool ConnectBlock(const CBlock& block, CValidationState& state, CBlockIndex* pin
         return true;
     }
 
-    bool fScriptChecks = pindex->nHeight >= Checkpoints::GetTotalBlocksEstimate();
+    bool fScriptChecks = true;
+    CBlockIndex *pindexLastCheckpoint = Checkpoints::GetLastCheckpoint();
```

Comment:
This is the part that confuses me. It looks like the current block being evaluated never appears in this new code block.